driver-core/src/test/java/com/datastax/driver/core/CCMBridge.java

```diff
@@ -941,6 +980,12 @@ public void run() {
             return ccm;
         }
 
+        private static boolean isThriftSupported(boolean dse, VersionNumber version) {
+            return
+                    (!dse && version.compareTo(VersionNumber.parse("3.10")) < 0)
```

Comment:
I think C* 4.0 will be the first version without thrift.  It will still be present in 3.10, but not in trunk.